synapse/lib/cell.py

```diff
@@ -4414,6 +4434,111 @@ async def ps(self, user):
 
         return retn
 
+    async def getAhaProxy(self, timeout=None, feats=None):
+
+        if self.ahaclient is None:
+            return
+
+        proxy = await self.ahaclient.proxy(timeout=timeout)
+        if proxy is None:
+            logger.warning('AHA client connection failed.')
+            return
+
+        if feats is not None:
+            for name, vers in feats:
+                if not proxy._hasTeleFeat(name, vers):
+                    logger.warning(f'AHA server does not support feature: {name} >= {vers}')
+                    return None
+
+        return proxy
+
+    async def callPeerApi(self, todo, timeout=None):
+        '''
+        Yield responses from our peers via the AHA gather call API.
+        '''
+        proxy = await self.getAhaProxy(timeout=timeout, feats=(feat_aha_callpeers_v1,))
+        if proxy is None: # pragma: no cover
+            return
```

Comment:
We should discuss whether callPeerApi is correct to return `None` or raise. A caller who expects a tuple of `isok, valu` results and gets a single `None` object back can easily end up generating `TypeError` when unpacking or accessing a response item by index value.

Reply:
This API does not return `None`. It is a generator and so we are yielding no results.